calculator.js

```diff
@@ -1,28 +1,61 @@
 
-const exampleAdditionInput = {
-  num1: 3,
-  num2: 5,
-  operation: 'add',
-}
-// '3 + 5 = 8'
+const calculateAdd = function(input) {
+  let result = input.num1 + input.num2
+  console.log(`${input.num1} + ${input.num2} = ${result}`);
+   return;
+};
+const calculateSub = function(input ){
+  let result = input.num1 - input.num2
+  return console.log(`${input.num1} - ${input.num2} = ${result}`);
+  return;
+};
+const calculateDiv = function(input) {
+  if (input.num1 === 0 || input.num2 === 0){
+  console.log('You cannot divide by zero');
+   return;
+  }
+  let result = input.num1 / input.num2
+  console.log(`${input.num1} / ${input.num2} = ${result}`);
+   return;
+};
+const calculateMult = function(input) {
+  let result = input.num1 * input.num2
+  console.log(`${input.num1} * ${input.num2} = ${result}`);
+  return;
+  };
 
-const calculateUserInput = function (error, promptInput) {
-  console.log('This is the value of the promptInput variable that got passed in by prompt, after our prompt package collect user input', promptInput);
-
-  // Questions to ask and answer:
-  // What is promptInput?
-  // What data type? What does it hold? What does it represent?
-  // How do we read values from it? What syntax?
-  // How can we use it?
-  // Can we call our existing functions now, inside of this function?
-}
-
-// Example manual testing of calculator.  
-calculateUserInput({}, {
-  num1: 3,
-  num2: 4,
-  operation: 'add',
-});
+const notANumber = function(input) {
+	const validatedInput = {operation: input.operation}
+  validatedInput.num1 = parseInt(input.num1)
+  validatedInput.num2 = parseInt(input.num2)
+  if (isNaN(validatedInput.num1)){
+    console.log(`${input.num1} is not a number.`);
+    return null;
+  };
+  if (isNaN(validatedInput.num2)){
+    console.log(`${input.num2} is not a number.`);
+    return null;
+  };
+  operatorCheck(validatedInput)
+};
 
+const operatorCheck = function(input){
+  if (input.operation === '+' || input.operation === 'add'){
+    return calculateAdd(input);
+};
+  if (input.operation === '-' || input.operation === 'subtract'){
+    return calculateSub(input);
+  };
+  if (input.operation === '/' || input.operation === 'divide'){
+    return calculateDiv(input);
+  };
+  if (input.operation === '*' || input.operation === 'multiply'){
+    return calculateMult(input);
+  };
+	return console.log('unsupported operator')
+};
```

Comment:
Rather than writing separate `if` statements, it is best to use a chained conditional with `if ... else if ... else if ...` so that multiple tests do not need to be performed by your computer.

```suggestion
  if (input.operation === '+' || input.operation === 'add'){
    return calculateAdd(input);
} else if (input.operation === '-' || input.operation === 'subtract'){
    return calculateSub(input);
  } else if (input.operation === '/' || input.operation === 'divide'){
    return calculateDiv(input);
  } else if (input.operation === '*' || input.operation === 'multiply'){
    return calculateMult(input);
  };
	return console.log('unsupported operator')
};
```

Reply:
You might also consider using case/switch.